### PR TITLE
HPCC-20190 MySQL cache options missing from roxie.xsd

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
None

### CODE REVIEW

**Comment by richardkchapman** [comment]
@rpastrana Please review

File: initfiles/componentfiles/configxml/roxie.xsd.in [hunk 1]
```diff
@@ -1211,6 +1211,27 @@
         </xs:appinfo>
       </xs:annotation>
     </xs:attribute>
+    <xs:attribute name="mysqlCacheCheckPeriod" type="xs:boolean" use="optional" default="10000">
```
Comment (rpastrana) [inline initfiles/componentfiles/configxml/roxie.xsd.in]: All 3 attributes are declared as xs:boolean, but they seem to be numeric

**Review by rpastrana — COMMENTED** [review]
@richardkchapman 1 minor comment.
@kenrowland let's make sure this change is captured in the config2.0 xsds as well.

File: initfiles/componentfiles/configxml/roxie.xsd.in (hunk 1 above)
Comment (richardkchapman) [reply initfiles/componentfiles/configxml/roxie.xsd.in]: Oops, too much cut/paste! I will fix

**Comment by richardkchapman** [comment]
@rpastrana Changed booleans to nonNegativeInteger

**Review by rpastrana — APPROVED** [review]
@richardkchapman looks reasonable to me.

**Comment by richardkchapman** [comment]
@ghalliday Please merge

**Comment by ghalliday** [comment]
Should there be an entry in roxie.xsd as well as roxie.xsd.in?  What is the relationship between those two files?

**Pull request merged**